synapse/config/server.py

```diff
@@ -368,14 +368,14 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
 
         # Whether to enable user presence.
         presence_config = config.get("presence") or {}
-        self.use_presence = presence_config.get("enabled")
-        if self.use_presence is None:
-            self.use_presence = config.get("use_presence", True)
-
-        # Selectively enable syncing of presence, even if it is disabled.
-        self.use_presence_for_sync = presence_config.get(
-            "enabled_for_sync", self.use_presence
-        )
+        presence_enabled = presence_config.get("enabled")
+        if presence_enabled is None:
+            presence_enabled = config.get("use_presence", True)
+
+        # Whether presence is enabled *at all*.
+        self.presence_enabled = bool(presence_enabled)
+        # Whether to internally track presence, requires that presence is enabled,
+        self.track_presence = self.presence_enabled and presence_enabled != "untracked"
```

Comment:
I think `track_presence` is very clear---thank you!

I think the phrase "Whether presence is enabled" is still a bit confusing. Can we call `self.presence_enabled` something like `self.report_presence`? (I'm guessing that this bool now controls the reporting of presence down /sync and in response to explicit presence GETs).

Reply:
It ends up getting used a bit internally too, I'm not sure that `presence_enabled` is the best name, but I think `report_presence` doesn't quite capture it either.

So it gets used for:

1. Whether to calculate presence for sync & initial sync.
2. Whether to return real presence info from `/presence`.
3. Whether to persist any presence information into the database.
4. Whether the presence handler APIs can be called to update presence state.

Reply:
I think `report_presence` easily covers 1 & 2, it doesn't feel quite right for 3 & 4 though.

Reply:
Blimey, that's a lot to juggle. Let's leave it as-is then. If nothing else, we'll have these comments as signposts for our future selves!